api/keto/acl/v1/acl.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package keto.acl.v1;
+
+option go_package = "github.com/ory/keto/api/keto/acl/v1;acl";
+option csharp_namespace = "Ory.Keto.Acl.V1";
+option java_multiple_files = true;
+option java_outer_classname = "AclProto";
+option java_package = "sh.ory.keto.acl.v1";
+option php_namespace = "Ory\\Keto\\Acl\\V1";
+
+
+// RelationTuple relates an Object with a Subject.
+//
+// While a tuple reflects a relationship between Object
+// and Subject, they do not completely define the effective ACLs.
+message RelationTuple {
+  // The object related by this tuple.
+  Object object = 1;
+  // The relation between an Object and a Subject.
+  string relation = 2;
+  // The subject related by this tuple.
+  // A Subject either represents a concrete subject id or
+  // a SubjectSet that expands to more Subjects.
+  Subject subject = 3;
+  // Used to perform a consistent read-modify-write (lock).
+  string etag = 4;
+}
+
+// Object is an object in a namespace
+// referenced by a RelationTuple or SubjectSet.
+//
+// It represents a "resource" or "digital object".
+message Object {
+  // The namespace of the object.
+  // This effectively is the namespace of the whole RelationTuple.
+  string namespace = 1;
```

Comment:
Namespaces will have to have an integer ID that is always constant. This then allows to set an alias (i.e. name) that can be changed. I think it is good to have the alias here, what do you think? So completely fine as you defined it IMO.

Reply:
IMO I wouldn't necessarily allow and add aliasing to namespaces. A namespace's FQDN should be just as constant as a primary key.

Why do you want to key a namespace to an integer?

Reply:
I am pretty sure that people will want to rename namespaces... Having an integer ID allows to derive table names from that ID instead of the string name that people would like to change. This way we can have changes to the name but not to the actual table. Anyway, the API should probably use the string name/alias/... anyway, and whether we use integer IDs or not will be part of the namespace config PR.

Reply:
Okey, I see namespaces here quite like a k8s namespace or even database name that does not get changed and is referenced in code and/or configuration. How often would a namespace even change? Giving a namespace a future proof name internal to the service that uses Keto is our users responsibility to get right. Usually namespaces are not even customer-facing so it's pretty internal.